### PR TITLE
Move footer translations to translations

### PR DESCRIPTION
**Pull request by pablothedude**
Move the feedback footer translations to the translation files to
prevent excessive configuration in deploy through the parameters.yml.
That way also missing translations are logged when a translation is
missing and we get rid of the CompilerPass.

`https://www.pivotaltracker.com/story/show/162414108`

**Change base before merging!**

### CODE REVIEW

**Comment by pablothedude** [comment]
@thijskh I saw that for the pages below the configuration for the footer blocks were missing. Would you like me to add them now?

```
stepup-callout-unknown.html.twig
stepup-callout-unmet-loa.html.twig
stepup-callout-user-cancelled.html.twig
generic.html.twig
authn-context-class-ref-blacklisted.html.twig
```

**Comment by thijskh** [comment]
Yes

File: languages/messages.nl.php [hunk 1]
```diff
@@ -1,4 +1,8 @@
 <?php
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
```
Comment (MKodde) [inline languages/messages.nl.php:5]: Did you add these by accident?

File: src/OpenConext/EngineBlockBundle/Resources/config/routing/feedback.yml [hunk 2]
```diff
@@ -1,139 +1,144 @@
+feedback_unknown_error:
+    path: '/feedback/unknown-error'
+    methods: [GET]
+    defaults: { _controller: engineblock.controller.authentication.feedback:unknownErrorAction }
```
Comment (MKodde) [inline src/OpenConext/EngineBlockBundle/Resources/config/routing/feedback.yml:4]: This change seems unrelated to the PR. It looks to be boyscout work and I love what you did here. Please consider moving this change and the ones related to it into a separate commit. That way the commit history regarding this change is better contextualized.

File: src/OpenConext/EngineBlockBundle/Twig/Extensions/Extension/Feedback.php [hunk 3]
```diff
@@ -160,12 +167,4 @@ private function retrieveFeedbackInfo()
...
         return $feedbackInfoMap;
     }
-
-    private function convertTemplateName($templateName)
-    {
-        $template = end(explode('/', $templateName));
-        $stripped = array_shift(explode('.', $template));
-        $convertedDashes = str_replace('-', '_', $stripped);
-        return $convertedDashes;
-    }
```
Comment (MKodde) [inline src/OpenConext/EngineBlockBundle/Twig/Extensions/Extension/Feedback.php:170]: :wave: bye bye ugly code block!

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Features/ClearErrorMessages.feature [hunk 4]
```diff
@@ -299,9 +299,6 @@ Feature:
      And I give my consent
     Then I should see "Error - The Assertion is not yet valid or has expired"
      And I should see ART code "44601"
-      # The Wiki link and IdP support mailto link should be visible on this page.
-     And I should see "OpenConext Wiki"
-     And I should see "support@openconext.org"
```
Comment (MKodde) [inline src/OpenConext/EngineBlockFunctionalTestingBundle/Features/ClearErrorMessages.feature:304]: Why remove this? Did a requirement change to no longer display the wiki link in this context?

File: app/config/parameters.yml.dist [hunk 5]
```diff
@@ -233,70 +233,3 @@ parameters:
...
-
-    ##########################################################################################
-    ## ERROR PAGES
-    ##########################################################################################
-    ## This is the configuration to be used on the error feedback pages.
-    error_feedback_wiki_links:
-        fallback:
-            en: '%openconext_support_url%'
-            nl: '%openconext_support_url%'
-            pt: '%openconext_support_url%'
```
Comment (MKodde) [inline app/config/parameters.yml.dist:246]: By removing the fallback feature, how can you now ensure the default wiki link is displayed when you did not provide a `specified` entry for your error page?

File: languages/messages.nl.php (hunk 1 above)
Comment (pablothedude) [reply languages/messages.nl.php:5]: :+1: removed

File: app/config/parameters.yml.dist (hunk 5 above)
Comment (pablothedude) [reply app/config/parameters.yml.dist:246]: Then you could provide a `fallback` entry instead. This will have the added benefit that we don't have to maintain the logic and you could even refrain from using a wiki page if you don't see fit.

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Features/ClearErrorMessages.feature (hunk 4 above)
Comment (pablothedude) [reply src/OpenConext/EngineBlockFunctionalTestingBundle/Features/ClearErrorMessages.feature:304]: This was the only test for error messages in the footer blocks. I'll look into a dedicated test to up the coverage.

File: src/OpenConext/EngineBlockBundle/Resources/config/routing/feedback.yml (hunk 2 above)
Comment (pablothedude) [reply src/OpenConext/EngineBlockBundle/Resources/config/routing/feedback.yml:4]: Will look into that

Comment (pablothedude) [reply src/OpenConext/EngineBlockBundle/Resources/config/routing/feedback.yml:4]: Done: https://github.com/OpenConext/OpenConext-engineblock/commit/c8f77b6cf6e024b856b7767067620ee574db98a8

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Features/ClearErrorMessages.feature (hunk 4 above)
Comment (pablothedude) [reply src/OpenConext/EngineBlockFunctionalTestingBundle/Features/ClearErrorMessages.feature:304]: Added: https://github.com/OpenConext/OpenConext-engineblock/commit/04b6819880588994e47f113f86458b40300761cb

File: src/OpenConext/EngineBlockBundle/Resources/config/routing/feedback.yml (hunk 2 above)
Comment (MKodde) [reply src/OpenConext/EngineBlockBundle/Resources/config/routing/feedback.yml:4]: Thanks for isolating this change!

File: app/config/parameters.yml.dist (hunk 5 above)
Comment (pablothedude) [reply app/config/parameters.yml.dist:246]: Validated

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Features/FeedbackFooters.feature [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      And I should not see "OpenConext Wiki"
+      And I should see "Helpdesk"
+      And I should not see "support@openconext.org"
+
+
+  Scenario: When a IdP specific error page is shown and a translation is configured the support emailaddress of the IdP should be visible
+    Given The clock on the IdP "Dummy Idp" is ahead
+    And I have configured the following translations:
+      | Key                                                                                       | Value                          |
+      | error_feedback_idp_contact_label_small_authentication_feedback_response_clock_issue       | MAIL                           |
```
[the remark this answers is not in the document]
Comment (pablothedude) [reply src/OpenConext/EngineBlockFunctionalTestingBundle/Features/FeedbackFooters.feature:38]: I've thought about that but I found testing the large viewport label `support@openconext.org` sufficient because this didn't seem to be implemented as mobile first.
